### ISSUE TITLE
rancher-desktop WSL distribution has DNS resolution issues through host machine when Windows' Hyper-V feature is enabled

### ISSUE THREAD

**Issue by jbaum517** [rancher-sandbox/rancher-desktop#2557]
### Actual Behavior

When I had Hyper-V enabled on my machine, I could not ever get Rancher Desktop to startup successfully. After launching the WSL distributions, setting up all the dependencies, etc, eventually it would be "Waiting for Services" for a long while and then eventually it would give a popup like "Error Starting Kubernetes" - "Client network socket disconnected before secure TLS connection was established".

### Steps to Reproduce

1. Turn on Hyper-V
2. Restart your machine
3. Install the latest version of RD
4. Start up the app
5. Wait for it to try to setup everything - get an error
6. You can also attempt something like "wsl -d rancher-desktop nslookup google.com" which will timeout due to dns issues.

### Result

Then receive a failure of "Error Starting Kubernetes" - "Client network socket disconnected before secure TLS connection was established".

### Expected Behavior

No DNS issues, clean install and RD startup of app and wsl distributions without kubernetes errors.

### Rancher Desktop Version

1.4.1

### Rancher Desktop K8s Version

1.23.8

### Which container runtime are you using?

moby (docker cli)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 10 - OS build 19042.1706

### What CPU architecture are you using?

x64

### Windows User Only

I am using a corporate VPN through zscaler. There may also be some security and firewall settings preconfigured on this work laptop but i did not configure anything special.

**Comment by byjrack** [comment]
You may want to see if adjusting the MTU has an effect.  The layers of virtualization and network tunnels can lead to packet fragmentation that can affect the TLS handshake.

`sudo ip link set mtu 1250 dev eth0`

You can include that as [part of a provisioning script](https://docs.rancherdesktop.io/how-to-guides/provisioning-scripts) if it works.

**Comment by andrew-landsverk-win** [comment]
I get this error even with Hyper V disabled. I had to write a provisioning script to manually set /etc/resolv.conf to point to our nameservers and then docker pull and other commands started working.

**Comment by byjrack** [comment]
Yup good point DNS is another cause I have seen of TLS failure.  @andrew-landsverk-win did you try the experimentalResolver feature to see if that made a difference in your case?  

As a note even w Hyper-V disabled you are still using the Virtualization Platform features which include the virtual switching.  You can see when you start a WSL2 distro the "WSL" switch show up in virtmgmt.msc and Get-VMSwitch.

**Comment by andrew-landsverk-win** [comment]
@byjrack I have not yet tried the experimentalResolver.

**Comment by jandubois** [comment]
Multiple DNS fixes have shipped since this was filed against v1.4.1: the host resolver was stabilized in [v1.5.0](https://github.com/rancher-sandbox/rancher-desktop/releases/tag/v1.5.0), truncated DNS responses were fixed in [v1.6.0](https://github.com/rancher-sandbox/rancher-desktop/releases/tag/v1.6.0), and VPN-specific DNS issues were addressed in [v1.17.1](https://github.com/rancher-sandbox/rancher-desktop/releases/tag/v1.17.1) and [v1.18.0](https://github.com/rancher-sandbox/rancher-desktop/releases/tag/v1.18.0). If you still see DNS resolution failures with Hyper-V and a corporate VPN on a current release, please open a new issue with version details and diagnostic output — the DNS stack has changed significantly.
